### PR TITLE
Add Nns

### LINKED ISSUE #1545: NFT assets in NNS scenarios
**Summary or problem description**

NGD is designing the NNS system. In NNS, domain names have ownership and independence, which is in line with [NFT](https://github.com/neo-project/proposals/pull/41) application scenarios.

**Need to discuss** ：

1. Whether it is necessary to design according to the NFT standard in NNS?
2. Whether the existing NFT standard is enough? 

**Neo Version**
- Neo 3

**Where in the software does this update applies to?**
- Other:NNS

### PR DESCRIPTION
**Pull request by doubiliu**
close #1545 Pre-PR #1670  is required.

In this PR, we have completed an NNS implementation.

**Where in the software does this update applies to?**
- CLI
- Plugins
- RPC (HTTP)
- SDK

### CODE REVIEW

**Comment by shargon** [comment]
Could you ceate two PR, one for NEP11 and other for NS?

**Comment by Tommo-L** [comment]
@doubiliu will help split it.

**Comment by doubiliu** [comment]
Already split

File: src/neo/SmartContract/Native/Nns/NnsContract.cs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+            return engine.Snapshot.Storages.TryGet(CreateStorageKey(Prefix_Roots)).GetInteroperable<RootDomainState>().Roots;
+        }
+
+        [ContractMethod(1_03000000, ContractParameterType.Boolean, CallFlags.AllowModifyStates, ParameterTypes = new[] { ContractParameterType.ByteArray, ContractParameterType.Hash160 }, ParameterNames = new[] { "name", "owner" })]
+        private StackItem Register(ApplicationEngine engine, Array args)
+        {
+            byte[] tokenId = args[0].GetSpan().ToArray();
+            string name = Encoding.UTF8.GetString(tokenId).ToLower();
+            string[] domains = name.Split(".");
+            if (domains.Length > 3) return false; // only the root and first-level name can be registered
```
Comment (joeqian10) [inline src/neo/SmartContract/Native/Nns/NnsContract.cs:54]: I think here should be `if (domains.Length > 2) return false;`.

Comment (Tommo-L) [reply src/neo/SmartContract/Native/Nns/NnsContract.cs:54]: ?

**Pull request closed without merging**